src/docs/product/integrations/integration-platform/public-integration.mdx

```diff
@@ -0,0 +1,156 @@
+---
+title: Public Integrations
+sidebar_order: 1
+description: "Learn more about Sentry's public integrations."
+---
+
+Public integrations are for the "general public" of Sentry users. These start in an unpublished state for development purposes and can be submitted later for approval to publish. For more information, check out the [section on Publication](#publication).
+
+The code examples in the sections below demonstrate a potential use-case that involves a Flask app receiving new issue webhooks from Sentry, calling the Sentry API for more data about the issue, and pushing it to Pushover as a generator of desktop/mobile notifications.
+
+## Installation
+
+Users will have the option to install your integrations on the **Integrations** page in [sentry.io](https://sentry.io). If your integration is still in an unpublished state, the only Sentry organization that will be able to see or install it will be the organization that created the integration. Once published, clicking the name of the integration will allow users to see a description of your integration and the permissions that will be granted if the user chooses to install.
+
+![A list of integrations that a user can install.](integration-directory.png)
+
+## OAuth Process
+
+After installation, if your user has approved all permissions, Sentry will generate a grant code and an installation ID. This information is sent by way of the `installation.created` webhook to the webhook URL specified in your configuration.
+
+However, if your integration has a redirect URL configured, the integration redirects the user’s browser to the configured URL with the grant code and installation ID in the query params.
+
+Start building your integration by implementing the redirect URL endpoint, `/setup` — typically where you exchange the grant code for a token that allows you to make API calls to Sentry:
+
+```python
+import requests
+from flask import redirect, request
+
+@app.route('/setup', methods=['GET'])
+def setup():
+    code = request.args.get('code')
+    install_id = request.args.get('installationId')
+
+    url = u'https://sentry.io/api/0/sentry-app-installations/{}/authorizations/'
+    url = url.format(install_id)
+
+    payload = {
+        'grant_type': 'authorization_code',
+        'code': code,
+        'client_id': 'your-client-id',
+        'client_secret': 'your-client-secret',
+    }
+
+    resp = requests.post(url, json=payload)
+    data = resp.json()
+
+    token = data['token']
+    refresh_token = data['refreshToken']
+    # ... Securely store the install_id, token and refresh_token in DB ...
+
+    return redirect('https://sentry.io/settings/')
+```
+
+## Auth Tokens
+
+Public integrations generate [Authentication Tokens](/product/integrations/integration-platform/#auth-tokens) after a successful [OAuth Installation](#oauth-process).
+These tokens automatically expire every eight hours, meaning they must be refreshed manually.
+
+
+## Refreshing Tokens
```

Comment:
Since this has to do with auth tokens, does it make sense to make this a sub-section of that?

```suggestion
### Refreshing Tokens
```